src/packtPublishingFreeEbook.py

```diff
@@ -141,6 +142,13 @@ def grab_ebook(self, api_client):
         offer_id = offer_data.get('id')
         product_id = offer_data.get('productId')
 
+        product_response = api_client.get(PACKT_PRODUCT_SUMMARY_URL.format(product_id=product_id))
+        self.book_data = {'id': product_id, 'title': product_response.json()['title']}\
+            if product_response.status_code == 200 else None
+
+        if self.cfg.anticaptcha_clientkey is None:
+            return
```

Comment:
We cannot return `None` silently. What if the user won't put AntiCaptcha API key in the configuration file and run script with the `-gd` option, than grabbing will fail, `grab_ebook` will return `None` and the script will break as it won't be able to find the e-book to be downloaded.

Reply:
It's already breaking now but in the more general case of not allowing the script to run at all even if no anti-captcha config should be required.  In the case you give, the script will still fail, just at a later point, so the user will still be notified that the grab and download failed.

If you think it would be better to provide another command line option whereby it is possible to only download todays item if it is already claimed, I'd be all for that.  This patch is just meant to resolve the issue of the script requiring configuration that isn't needed for any operation at all.